mesop/web/src/services/channel.ts

```diff
@@ -196,6 +202,13 @@ export class Channel {
             break;
           }
           case UiResponse.TypeCase.ERROR:
+            if (
+              uiResponse.getError()?.getException() ===
+              'No state hash found in state session backend.'
+            ) {
+              // How to retry?
+              // this.init(this.initParams, request);
+            }
```

Comment:
Yeah, it didn't work for me. The retry would get "cancelled" in the network tab. So it would never complete. Not sure why. But I didn't dig too deep yet.

Reply:
I think this could be because if you do `this.init` here, the original EventSource hasn't closed yet, and in L104 https://github.com/google/mesop/blob/c7aefdc62eb040d6d2f0a93ba253d0e46d2f6ce6/mesop/web/src/services/channel.ts#L104 it'll close the EventSource *after* you've done the retry with this.init (which is why I think you saw the retry get cancelled).

Instead if you put this in this.queuedEvents (probably the front of the queue), similar to https://github.com/google/mesop/blob/c7aefdc62eb040d6d2f0a93ba253d0e46d2f6ce6/mesop/web/src/services/channel.ts#L231, then it should work

Reply:
Nice adding it to the queue worked.  I think the only thing left here is handling the case where the retry keeps failing, so we probably want to give up after a couple failures.

Reply:
Actually, I think it's OK. There should never be an infinite loop since the retry includes the state. In the worst case the memory cache is not working at all and every request takes two tries.